### DISCUSSION TITLE
postPersist triggered after or before database operations ?

### DISCUSSION THREAD

**Discussion by sdev-maxime** [doctrine/doctrine2#9831] [Support Questions] [answered]
Hi, I'm searching  an Event triggered after database operations on New insert, cause the postUpdate is only on already existing data.

So I've found this:

![image](https://user-images.githubusercontent.com/52406045/172629563-23913fb3-fad9-49c7-9b38-ce4b59a59858.png)

**_"It will be invoked after the database insert operations"_**
But in fact, if I die()/stop my script in the listener, my database isn't updated. Is it a mistake on the documentation ?

What is the real behavior of the postPersist() event ? trigger an event on new entity **after** database operation or trigger an event **before** flush operation ?
This is confusing, because the EntityManager->persist() operation has same name too but seems different to postPersist().

Doctrine/orm 2.9
Symfony 5.4
Documentation: https://www.doctrine-project.org/projects/doctrine-orm/en/2.9/reference/events.html

**Comment by derrabus** [comment, marked as answer]
The `postPersist` event is triggered after the `INSERT` operations, but before the transaction is committed. If your application dies during the event, the database will roll back your changes.

**Reply by sdev-maxime** [reply]
Ok, I understand better now.  Thanks for the fast answer @derrabus.
